config/template.go

```diff
@@ -57,6 +57,39 @@ func parseEnvironment(environ []string) Environment {
 	return env
 }
 
+func envFunc(env string) string {
+	return os.Getenv(env)
+}
+
+// loop accepts 1 or two parameters
+// loop 5 returns 0 1 2 3 4 or loop 5 8 returns 5 6 7 or loop 5 1 returns 5 4 3 2
+func loop(params ...int) ([]int, error) {
```

Comment:
This `loop` function is different than [the one in `consul-template`](https://github.com/hashicorp/consul-template/blob/92746fc5cf86dbb113558bacec43459a65c8df14/template/funcs.go#L663-L685). But if I'm reading everything right this might mean that we have the added flexibility of having something like `{{ range $i, $e := loop 5 8 }}` work.

Can you demonstrate in tests how this function would be consumed? i.e. can you show in the test a couple examples of using `range` with this?